### PR TITLE
PYTHON-3396 Support the Azure VM-assigned Managed Identity for Automatic KMS Credentials

### PR DESCRIPTION
**Pull request by blink1073**
Will require a corresponding integration test in mongo-python-driver

### CODE REVIEW

File: bindings/python/pymongocrypt/credentials.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+except ImportError:
+    _HAVE_AUTH_AWS = False
+
+from pymongocrypt.errors import MongoCryptError
+
+import requests
+
+
+_azure_creds = namedtuple("_azure_creds", ["access_token", "expires_in"])
+_azure_creds_cache = None
```
Comment (ShaneHarvey) [inline bindings/python/pymongocrypt/credentials.py:30]: Why do we cache azure creds but not GCP creds? Is that coming later?

File: bindings/python/pymongocrypt/credentials.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            creds_dict["sessionToken"] = aws_creds.token
+        creds['aws'] = creds_dict
+    if on_demand_gcp:
+        creds['gcp'] = _get_gcp_credentials()
+    if on_demand_azure:
+        try:
+            creds['azure'] = _get_azure_credentials()
+        except Exception as e:
+            _azure_creds_cache = None
+            raise e
```
Comment (ShaneHarvey) [inline bindings/python/pymongocrypt/credentials.py:135]: I believe it's preferred to use bare raise instead so that the traceback is unadulterated.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: bindings/python/pymongocrypt/credentials.py (hunk 1 above)
Comment (blink1073) [reply bindings/python/pymongocrypt/credentials.py:30]: In the scope document for this issue it was determined that caching was not needed for GCP creds.

File: bindings/python/pymongocrypt/credentials.py (hunk 2 above)
Comment (blink1073) [reply bindings/python/pymongocrypt/credentials.py:135]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**